src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/ExistingResourceValidator.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Linq;
+using System.Threading.Tasks;
+using Amazon.CloudControlApi.Model;
+using AWS.Deploy.Common.Data;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    public class ExistingResourceValidator : IOptionSettingItemValidator
+    {
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+
+        /// <summary>
+        /// The Cloud Control API resource type that will be used to query Cloud Control API for the existance of a resource.
+        /// </summary>
+        public string? ResourceType { get; set; }
+
+        public ExistingResourceValidator(IAWSResourceQueryer awsResourceQueryer)
+        {
+            _awsResourceQueryer = awsResourceQueryer;
+        }
+
+        public async Task<ValidationResult> Validate(object input)
+        {
+            if (string.IsNullOrEmpty(ResourceType))
+                throw new MissingValidatorConfigurationException(DeployToolErrorCode.MissingValidatorConfiguration, $"The validator of type '{typeof(ExistingResourceValidator)}' is missing the configuration property '{nameof(ResourceType)}'.");
+            var resourceName = input.ToString();
+            if (string.IsNullOrEmpty(resourceName))
+                return ValidationResult.Failed($"The resource name is empty and cannot be validated.");
+
+            switch (ResourceType)
+            {
+                case "AWS::ElasticBeanstalk::Application":
+                    var beanstalkApplications = await _awsResourceQueryer.ListOfElasticBeanstalkApplications(input.ToString());
+                    if (beanstalkApplications.Any(x => x.ApplicationName.Equals(input.ToString())))
+                        return ValidationResult.Failed($"An Elastic Beanstalk application already exists with the name '{input.ToString()}'.");
+                    break;
+
+                case "AWS::ElasticBeanstalk::Environment":
+                    var beanstalkEnvironments = await _awsResourceQueryer.ListOfElasticBeanstalkEnvironments(environmentName: input.ToString());
+                    if (beanstalkEnvironments.Any(x => x.EnvironmentName.Equals(input.ToString())))
+                        return ValidationResult.Failed($"An Elastic Beanstalk environment already exists with the name '{input.ToString()}'.");
```

Comment:
If the resource is in a terminated state, this could be confusing to users, because they may not be aware it is in their account. Consider adding the state value to the message for Beanstalk Env and App